rust/utils/run-locally/src/main.rs

```diff
@@ -591,189 +483,31 @@ fn termination_invariants_met(num_expected_messages: u32) -> Result<bool> {
-        .open(p)
-        .expect("Failed to open file")
-}
-
-fn build_cmd(
-    cmd: &[&str],
-    log: impl AsRef<Path>,
-    log_all: bool,
-    wd: Option<&str>,
-    env: Option<&HashMap<&str, &str>>,
-    assert_success: bool,
-) {
-    assert!(!cmd.is_empty(), "Must specify a command!");
-    let mut c = Command::new(cmd[0]);
-    c.args(&cmd[1..]);
-    if log_all {
-        c.stdout(Stdio::inherit());
-    } else {
-        c.stdout(append_to(log));
-    }
-    if let Some(wd) = wd {
-        c.current_dir(wd);
-    }
-    if let Some(env) = env {
-        c.envs(env);
-    }
-    let status = c.status().expect("Failed to run command");
-    if assert_success {
-        assert!(
-            status.success(),
-            "Command returned non-zero exit code: {}",
-            cmd.join(" ")
-        );
-    }
-}
-
-fn make_static(s: String) -> &'static str {
-    Box::leak(s.into_boxed_str())
-}
-
-fn spawn_cmd_with_logging(
-    name: &str,
-    mut command: Command,
-    log_prefix: &'static str,
-    log_all: bool,
-    log_dir: &PathBuf,
-) -> (std::process::Child, JoinHandle<()>, JoinHandle<()>) {
-    println!("Spawning {}...", name);
-    let mut child = command
-        .spawn()
-        .unwrap_or_else(|_| panic!("Failed to start {}", name));
-    let stdout_path = concat_path(log_dir, format!("{}.stdout.log", log_prefix));
-    let child_stdout = child.stdout.take().unwrap();
-    let stdout = spawn(move || {
-        if log_all {
-            prefix_log(child_stdout, log_prefix)
-        } else {
-            inspect_and_write_to_file(
-                child_stdout,
-                stdout_path,
-                &["ERROR", "message successfully processed"],
-            )
-        }
-    });
-    let stderr_path = concat_path(log_dir, format!("{}.stderr.log", log_prefix));
-    let child_stderr = child.stderr.take().unwrap();
-    let stderr = spawn(move || {
-        if log_all {
-            prefix_log(child_stderr, log_prefix)
-        } else {
-            inspect_and_write_to_file(child_stderr, stderr_path, &[])
-        }
-    });
-    (child, stdout, stderr)
+fn kill_scraper_postgres(build_log: impl AsRef<Path>, log_all: bool) {
+    build_cmd(
+        ProgramArgs::new("docker")
+            .cmd("stop")
+            .cmd("scraper-testnet-postgres"),
+        &build_log,
+        log_all,
+        false,
+    )
+    .join();
 }
 
-fn run_agent(
-    name: &str,
-    env: &HashMap<&str, &str>,
-    args: &[&str],
-    log_prefix: &'static str,
-    log_all: bool,
-    log_dir: &PathBuf,
-) -> (std::process::Child, JoinHandle<()>, JoinHandle<()>) {
-    let mut command = Command::new(format!("target/debug/{}", name));
-    command
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .envs(env)
-        .args(args);
-    spawn_cmd_with_logging(name, command, log_prefix, log_all, log_dir)
+fn filter_anvil_logs(_log: &str) -> bool {
+    // for now discard all anvil logs
+    false
```

Comment:
should this be true? Interpreted this as whether anvil logs should be silenced

Reply:
It is a filter function, so true means keep, false means discard. Follows the pattern of https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter

I will update docs on that though.